infrastructure/simulation-tool/src/utils/number.utils.ts

```diff
@@ -0,0 +1,24 @@
+import { BigNumber, BigNumberish } from 'ethers';
+
+type RandomAmountGenerator = Generator<BigNumber>;
+
+const getRandomBigNumber = (min: BigNumberish, max: BigNumberish): BigNumber => {
+    const diff = BigNumber.from(1).mul(max).sub(min);
```

Comment:
```suggestion
    const diff = BigNumber.from(max).sub(min);
```

Reply:
This is rather annoying as the result of next is `IteratorResult<T, TReturn>`, the definition of which says to me that unless we call done, the type of `value` is `T`, where `T` is the first generic parameter to the `Generator` type.
I should have checked in the code, though. Now that I did, I can see that the value type is indeed `any`. 🤬


<img width="704" alt="Screenshot 2023-07-28 at 09 50 54" src="https://github.com/rsksmart/rif-rollup/assets/36888576/801c5582-23bf-47e5-8d9e-f593876687e5">
<img width="832" alt="Screenshot 2023-07-28 at 09 52 09" src="https://github.com/rsksmart/rif-rollup/assets/36888576/d1771816-fd47-498c-b9eb-4a883bbf514d">

Reply:
done